app/controllers/publishers_controller.rb

```diff
@@ -129,6 +130,40 @@ def verify
   # Shown after verification is completed to encourage users to submit
   # payment information.
   def verification_done
+    @publisher = current_publisher
+    @publisher.uphold_state_token = SecureRandom.hex(64)
+    @publisher.save!
+  end
+
+  def uphold_verified
```

Comment:
It is pretty inelegant to have to modify the redirect URL manually. However, Uphold requires that the redirect URL be HTTPS. One solution would be to locally run a proxy, but that's a hassle in its own way.